### ISSUE TITLE
missing ipopt_auxdata.m

### ISSUE THREAD

**Issue by rdzman** [jonathancurrie/OPTI#6]
If I understand correctly, the MEX interface in Ipopt 3.11.x and later removed support for `options.auxdata` and moved that functionality into the M-file `ipopt_auxdata.m`. I recently discovered that this file is missing from OPTI Toolbox, requiring users who need it to find it and install it separately from the Ipopt source distribution.

Could you please add it to OPTI Toolbox? In the Ipopt source distribution it is found in `Ipopt/contrib/MatlabInterface/ipopt_auxdata.m`.

FYI, I am the primary developer of [MATPOWER](https://github.com/MATPOWER/matpower/), a power systems analysis and optimization package. MATPOWER includes support for using Ipopt as a solver, and I've been pointing Windows users to OPTI Toolbox as the easiest way to obtain a  working Ipopt. And MATPOWER **does** require the `auxdata` support.

**Comment by jonathancurrie** [comment]
Hi Ray,

Thanks for bringing this to my attention. I've had a quick look at the `ipopt_auxdata` source and as far as I can see, this file is simply wrapping MATLAB function handles, e.g.:

`objective = @(x, auxdata) 2*x + auxdata`

into

`new_objective = @(x) objective(x, auxdata)`

Then passing `new_objective` to the original `ipopt` mex file. This is a fairly standard operation to do, and I would think could be written into MATPOWER, without the use of this auxillary file?

Let me know your thoughts.

Jonathan

**Comment by rdzman** [comment]
Yes, that is precisely the function of `ipopt_auxdata`. However, since it is now a standard part of the Matlab interface in the Ipopt distribution itself, added in order to preserve a feature in the older MEX interface, it seems strange to me to exclude it from OPTI Toolbox with the idea that those who need that functionality can re-implement it themselves.

In my view, by not including`ipopt_auxdata` OPTI Toolbox currently only does a partial installation of Ipopt for Matlab, requiring users who need that functionality to go to the Ipopt source to complete the installation.

Is there a reason not to simply include it?

**Comment by jonathancurrie** [comment]
Hi Ray,

Apologies I just got back to this now. I guess my hesitation stems from the fact the OPTI IPOPT interface is deliberately different from the official one (e.g. handles linear constraints differently, among other differences), and never required the auxiliary data file to function. 

However, I agree that it is a pain for other projects to interface to OPTI and the official IPOPT simultaneously, thus I have added the file to the distribution, now on the master branch:
https://github.com/jonathancurrie/OPTI/commit/a220c469365ceb39e0d815cb886fef5d2b7b2cdc

Have a great Christmas!
Jonathan
